packages/cli/src/commands/target/list.ts

```diff
@@ -0,0 +1,136 @@
+import chalk from 'chalk';
+import ms from 'ms';
+import table from '../../util/output/table';
+import Client from '../../util/client';
+import { getCommandName } from '../../util/pkg-name';
+import type { ProjectLinked } from '@vercel-internals/types';
+import type { CustomEnvironment } from '../../util/target/types';
+
+export default async function list(
+  client: Client,
+  argv: any,
+  args: string[],
+  link: ProjectLinked
+) {
+  const { output } = client;
+  if (args.length !== 0) {
+    output.error(
+      `Invalid number of arguments. Usage: ${chalk.cyan(
+        `${getCommandName('target ls')}`
+      )}`
+    );
+    return 2;
+  }
+
+  const start = Date.now();
+  const projectUrl = `https://vercel.com/${link.org.slug}/${link.project.name}`;
+  const projectSlugLink = output.link(
+    chalk.bold(`${link.org.slug}/${link.project.name}`),
+    projectUrl,
+    {
+      fallback: () => chalk.bold(`${link.org.slug}/${link.project.name}`),
+      color: false,
+    }
+  );
+
+  output.spinner(`Fetching custom environments for ${projectSlugLink}`);
+
+  let url = `/projects/${encodeURIComponent(
+    link.project.id
+  )}/custom-environments?limit=20`;
+
+  const next = argv['--next'] || false;
+  if (next) {
+    url += `&until=${next}`;
+  }
+
+  const result: CustomEnvironment[] = await client.fetch(url, {
+    method: 'GET',
+    accountId: link.org.id,
+  });
+
+  output.stopSpinner();
+
+  const elapsed = ms(Date.now() - start);
+
+  result.unshift(
+    {
+      id: 'production',
+      slug: 'production',
+      createdAt: 0,
+      updatedAt: 0,
+      type: 'production',
+      description: '',
+      name: 'Production',
+      domains: [],
+    },
+    {
+      id: 'preview',
+      slug: 'preview',
+      createdAt: 0,
+      updatedAt: 0,
+      type: 'preview',
+      description: '',
+      name: 'Preview',
+      domains: [],
+    }
+  );
+
+  result.push({
+    id: 'development',
+    slug: 'development',
+    createdAt: 0,
+    updatedAt: 0,
+    type: 'development',
+    description: '',
+    name: 'Development',
+    domains: [],
+  });
```

Comment:
Non-blocking: 

1. Extract into function 

2. Change to make ordering more clear.
```
return [
  ...staticEnvironments,
  ...customEnvironments,
  ...developmentEnvironments
];
```